model/test_results.go

```diff
@@ -88,6 +89,51 @@ func (t *TestResults) Find(ctx context.Context) error {
 	return nil
 }
 
+// TestResultsFindOptions allows for querying with or without execution val
+type TestResultsFindOptions struct {
+	TaskID         string
+	Execution      int
+	EmptyExecution bool
+}
+
+// FindByTaskExecution searches the database for the TestResults associated with the provided task_id / execution pair.
+// The enviromemt should not be nil. If execution is empty, will default to most recent execution
```

Comment:
Do we need to actually mention the environment at all (it's misspelled btw)? I feel like it doesn't actually need to be stated since this is the working assumption in basically all code.